website/main/content/blog/2025-05-19-block-as-forest.md

```diff
@@ -0,0 +1,133 @@
+---
+title: The Unblocker - Blocks as a forest of trees
+date: 2025-05-19
+draft: false
+description: Unblocking progress through the design of blocks as a forest of trees.
+tags: [status-update, consensus]
+authors: [adlrocha]
+---
+
+I am pretty happy with the progress this week. Funnily, the main culprit for all of this progress
+has been the the work that Nazar has been doing on the definition of the block structure for our
+hierarchical consensus. I'll let Nazar dig deeper into what he's been doing here, but let me share
+in this post what this block structure entails, and how this has unblock several lines of work, and
+solved many issues for me.
+
+<!--more-->
+
+## Cracking the first piece of the puzzle: blocks as trees.
+
+If you recall from previous posts, we already had a pretty good sense of what is going to be the
+lifecycle of blocks in the system: from being created in a shard, to being submitted and committed
+to its parent, and how this information is implicitly propagated up and can be used to generate
+proofs about the history of a shard. Unfortunately, we were missing one piece to this puzzle that
+was dragging progress: making a decision of what would be the block structure, as this would
+influence the design of other mechanics in the system. After a few back-and-forths, it looks like
+Nazar has cracked it down on the PRs referenced by
+[this discussion](https://abundance.zulipchat.com/#narrow/channel/495788-research/topic/Mechanics.20of.20child.20block.20submission.20to.20parent.20chain/near/518994821).
+
+The key _"innovation"_ that this design for blocks has is that all the block information (including
+the header) is structured as trees. This has a lot of interesting properties, as it allows not only
+to leverage information from the header to generate proofs about the information included in the
+body, but also to generate proofs about the header itself without having to provide the whole
+header. This is a pretty powerful property for a hierarchical network architecture like the one we
+have, because it enables the generation of recursive proofs involving information from blocks (and
+implicitly state) belonging to different shards.
+
+Here's a high-level sketch from one of our discussion syncs that briefly depicts what this block as
+trees would look like (although I would recommend you to check the code to get a full sense of what
+is going on under the hood):
+
+<p align="center">
+<img alt="High-level sketch of the tree structure for blocks" src="block-header.png">
+</p>
+
+## Everything is a tree!
+
+Nazar already alluded to this in his post
+[Everything is a tree!](https://abundance.build/blog/2025-04-14-trees-everywhere/), but now it has
```

Comment:
Please use following URL, so it works locally, in production and will continue if/when we change the website URL:
```
../2025-04-14-trees-everywhere
```

Reply:
For some reason this didn't work the first time I tried, I retried and re-launched Hugo and now it works. Will be fixed in my next commit.